plenum/server/replicas.py

```diff
@@ -126,20 +131,23 @@ def num_replicas(self):
 
     @property
     def sum_inbox_len(self):
-        return sum(len(replica.inBox) for replica in self._replicas)
+        return sum(len(replica.inBox) for replica in self._replicas.values())
 
     @property
     def all_instances_have_primary(self) -> bool:
         return all(replica.primaryName is not None
-                   for replica in self._replicas)
+                   for replica in self._replicas.values())
 
     # TODO unit test
     @property
-    def primaries(self) -> list:
-        return [r.primaryName for r in self._replicas]
+    def primaries(self) -> dict:
+        primaries = dict()
+        for r in self._replicas.values():
+            primaries[r.instId] = r
```

Comment:
This property must return not replicas but names of instances primaries.

Reply:
Done